controllers/apps/v1beta3/emqx_handler.go

```diff
@@ -690,7 +665,10 @@ func generateAcl(instance appsv1beta3.Emqx, sts *appsv1.StatefulSet) (*corev1.Co
 	annotations["ACL/Base64EncodeConfig"] = base64.StdEncoding.EncodeToString([]byte(aclString))
 	sts.Annotations = annotations
 
-	container := sts.Spec.Template.Spec.Containers[0]
+	exist, container := findEmqxContainer(sts.Spec.Template.Spec.Containers)
+	if !exist {
+		return cm, sts
```

Comment:
return error, and do we have to check for errors? It looks like the `emqx` container will must exist

Reply:
my understanding is that the emqx container must exist

Reply:
i will check this error